### ISSUE TITLE
Allow setting a list of images to spawn from when users log in

### ISSUE THREAD

**Issue by yuvipanda** [pangeo-data/pangeo#87]
We'd like to:

1. Have a list of docker images specified in our config.yaml
2. Allow users to pick one of them when they launch.

I think the way to do this is:

1. Use http://zero-to-jupyterhub.readthedocs.io/en/latest/advanced.html#arbitrary-code-in-jupyterhub-config-py set extraConfigMap to have a list of docker images in config.yaml
2. Use `extraConfig` (from same link as 1) to make an [options form](http://jupyterhub.readthedocs.io/en/latest/reference/spawners.html?highlight=options#spawner-options-form). See https://github.com/google/kubeflow/blob/master/kubeflow/core/jupyterhub.libsonnet#L19 for an example of the Python code. You're subclassing KubeSpawner and adding more methods.

**Comment by mrocklin** [comment]
This would probably help people who have significantly different stacks than what we currently deploy.  In particular @rsignell-usgs and @jacobtomlinson and @niallrobinson

**Comment by yuvipanda** [comment]
This can happen easily once https://github.com/jupyterhub/kubespawner/pull/137 is merged, so I suggest we wait for it.

**Comment by jhamman** [comment]
Has anyone tried this yet? I suspect this is something that is going to interest a number of us.

**Comment by jacobtomlinson** [comment]
I'm looking at this again during the Pangeo developer meeting sprints. Help would be appreciated!

**Comment by jacobtomlinson** [comment]
It looks like we are waiting on `z2jh` 0.7 for this to make it's way upstream. However the functionality should be in `KubeSpawner` already so we may be able to add it manually for now.

**Comment by yuvipanda** [comment]
@jacobtomlinson 0.7 has been released a few weeks ago!

**Comment by rabernat** [comment]
We should also add options to select different resource options (CPU / GPU / memory).

**Comment by delgadom** [comment]
Could this feature be used to select preemptable/nonpreemptable nodes, or other configurable hardware options?

**Comment by rabernat** [comment]
profile_list! This is the technical solution

https://jupyterhub-kubespawner.readthedocs.io/en/latest/spawner.html?highlight=profile_list

**Comment by mrocklin** [comment]
I think that @rabernat's immediate need isn't so much with different
images, as it is with different resource configurations like high-memory or
GPU nodes.

**Comment by rabernat** [comment]
Correct. Now that we are customizing pangeo deployments, user-specified envs is not as high a priority. I am just interested in customizing resources. But I believe that the `profile_list` solution covers both use cases.

Perhaps I should have created a new issue after all.

**Comment by yuvipanda** [comment]
I *think* @dsludwig already sent a PR to hubploy to make it work with multiple images. I'll take a look later today
